test/check-application

```diff
@@ -994,13 +994,26 @@ class TestFiles(testlib.MachineCase):
         b.click("button.pf-m-primary")
         b.wait_in_text("[data-item='newdir1'] .item-owner", "testuser")
 
+        # Open folder in terminal from context menu
+        b.mouse("[data-item='newdir1']", "contextmenu")
+        b.click(".contextMenu button:contains('Open in terminal')")
+        b.enter_page("/system/terminal")
+        b.switch_to_top()
+        b.wait_js_cond('String(window.location.pathname) === "/system/terminal"')
+        b.wait_js_cond('window.location.hash === "#/?path=%2Fhome%2Fadmin%2Fnewdir1"')
+        b.go("/files")
+        b.enter_page("/files")
+        # Regular files should not have terminal feature
+        m.execute("echo 'some content' > /home/admin/newfile")
+        b.mouse("[data-item='newfile']", "contextmenu")
+        b.wait_not_present(".contextMenu button:contains('Open in terminal')")
```

Comment:
This is a race condition -- there won't be any menu immediately after the click. You first need to wait for a positive condition, like the "Delete" menu entry becoming visible.